### ISSUE TITLE
Autocorrect doesn't work in some cases for mark rule

### ISSUE THREAD

**Issue by micnguyen** [realm/SwiftLint#1837]
### New Issue Checklist

- [ ✅] Updated SwiftLint to the latest version
- [✅ ] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Bug Report
Some rules that have autocorrect according to README.md are not automatically applying any fixes. An example of this is the `mark` rule where it does not auto correct. When I change it to not auto correct, it successfully picks up the rule violation however.

```bash
$ swiftlint autocorrect --config .swiftlint_autocorrect.yml
```

### Environment
Swiftlint Cocoapods 0.21.0

```yml
# Explanation of rules: https://github.com/realm/SwiftLint/blob/master/Rules.md
whitelist_rules:
- closing_brace
- closure_spacing
- colon
- comma
- control_statement
- empty_enum_arguments
- empty_parameters
- empty_parenthesis_with_trailing_closure
- leading_whitespace
- legacy_cggeometry_functions
- legacy_constant
- legacy_constructor
- legacy_nsgeometry_functions
- mark
- opening_brace
- operator_usage_whitespace
- redundant_discardable_let
- redundant_optional_initialization
- return_arrow_whitespace
- protocol_property_accessors_order
- statement_position
- trailing_comma
- trailing_newline
- trailing_semicolon
- unused_closure_parameter
- vertical_whitespace
- void_return
excluded:
  - Pods
reporter: "xcode"
```

Xcode8.3.3

```swift
// This triggers a violation:
// Mark: UIPicker Delegate / Data Source
```

**Comment by marcelofabri** [comment]
@micnguyen Thanks for the issue! I was able to reproduce it with 0.22.0. Could you please open an issue for each rule that is not autocorrecting? FYI, I strongly believe that only some cases of those rules are not autocorrecting, so examples are welcome!

**Comment by Igor-Palaguta** [comment]
@marcelofabri I use swiflint 0.25.0. file_header rule is not autocorrected

**Comment by marcelofabri** [comment]
@Igor-Palaguta `file_header` doesn't support autocorrect.

**Comment by Igor-Palaguta** [comment]
@marcelofabri Thanks for response

**Comment by marcelofabri** [comment]
@micnguyen this seems to be fixed on the latest version. I'm closing this, feel free to reopen if it's still an issue.
